gtsfm/data_association/data_assoc.py

```diff
@@ -196,6 +204,68 @@ def run(
 
         return connected_data, data_assoc_metrics
 
+    def run_triangulation(
+        self,
+        cameras: Dict[int, gtsfm_types.CAMERA_TYPE],
+        tracks_2d: List[SfmTrack2d],
+    ) -> Tuple[List[Delayed], List[Delayed], List[Delayed]]:
+        """Performs triangulation of 2D tracks in parallel.
+
+        Refs:
+        - https://docs.dask.org/en/stable/delayed-best-practices.html#compute-on-lots-of-computation-at-once
+        - https://docs.dask.org/en/stable/delayed-best-practices.html#avoid-too-many-tasks
+
+        Args:
+            cameras: list of cameras wrapped up as Delayed.
+            tracks_2d: list of tracks wrapped up as Delayed.
+
+        Returns:
+            sfm_tracks: List of triangulated tracks.
+            avg_track_repoj_errors: List of average reprojection errors per track.
+            triangulation_exit_codes: exit codes for each triangulation call.
+        """
+
+        def triangulate_batch(point3d_initializer: Point3dInitializer, tracks_2d: List[SfmTrack2d]):
```

Comment:
Thanks for adding this Travis. nit: could we add a return type hint here?

Reply:
Done

Reply:
Marking as resolved, thanks!